lib/ansible/modules/cloud/amazon/elasticache_snapshot.py

```diff
@@ -0,0 +1,238 @@
+            response = {}
+            changed = False
+        else:
+            module.fail_json(msg="Unable to create the snapshot.", exception=traceback.format_exc(e))
+    return response, changed
+
+def copy(module, connection, name, target, bucket):
+    """ Copy an Elasticache backup. """
+    try:
+        response = connection.copy_snapshot(SourceSnapshotName=name,
+                                            TargetSnapshotName=target,
+                                            TargetBucket=bucket)
+        changed = True
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable to copy the snapshot.", exception=traceback.format_exc(e))
+    return response, changed
+
+def delete(module, connection, name):
+    """ Delete an Elasticache backup. """
+    try:
+        response = connection.delete_snapshot(SnapshotName=name)
+        changed = True
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "SnapshotNotFoundFault":
+            response = {}
+            changed = False
+        elif e.response['Error']['Code'] == "InvalidSnapshotState":
+            module.fail_json(msg="Error: InvalidSnapshotState. The snapshot is not in an available state or failed state to allow deletion."
+                             "You may need to wait a few minutes.")
+        else:
+            module.fail_json(msg="Unable to delete the snapshot.", exception=traceback.format_exc(e))
+    return response, changed
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(required=True, type='str'),
+            state=dict(required=True, type='str', choices=['present', 'absent', 'copy']),
+            replication_id=dict(type='str'),
+            cluster_id=dict(type='str'),
+            target=dict(type='str'),
+            bucket=dict(type='str'),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto required for this module')
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+    replication_id  = module.params.get('replication_id')
+    cluster_id = module.params.get('cluster_id')
+    target = module.params.get('target')
+    bucket = module.params.get('bucket')
+
+    # Retrieve any AWS settings from the environment.
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg = str("Either region or AWS_REGION or EC2_REGION environment variable or boto config aws_region or ec2_region must be set."))
+
+    connection = boto3_conn(module, conn_type='client',
+                            resource='elasticache', region=region,
+                            endpoint=ec2_url, **aws_connect_kwargs)
+
+    changed = False
+    response = {}
+
+    if state == 'present':
+        if not all((replication_id, cluster_id)):
```

Comment:
Oh. You're right, that's definitely not what's intended. PRs welcome! Is your cluster_id parameter == None?